.github/workflows/cli_actions.yml

```
@@ -34,7 +34,114 @@ jobs:
       - name: Run tests
         run: pub run test
         working-directory: metrics/cli/
-        
+
+  cli-build-actions:
+    name: Metrics CLI Build Actions
+    needs: [ cli-actions ]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ macos-latest, ubuntu-latest ]
+        include:
+          - os: macos-latest
+            output_name: "metrics_cli_macos"
+          - os: ubuntu-latest
+            output_name: "metrics_cli_linux"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+          ref: ${{ github.ref }}
+
+      - uses: cedx/setup-dart@v2
+        with:
+          release-channel: beta
+          version: 2.12.0-133.2.beta
+
+      - name: Get dependencies
+        run: pub get
+        working-directory: metrics/cli/
+
+      - name: Build Application
+        run: dart2native -o ${{ matrix.output_name }} bin/main.dart
+        working-directory: metrics/cli/
+
+      - name: Generate SHA256 Hash
+        run: shasum -a 256 metrics/cli/${{ matrix.output_name }} > metrics/cli/${{ matrix.output_name }}.sha256
+
+      - name: Upload SHA256 Hash
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.output_name }}.sha256
+          path: metrics/cli/${{ matrix.output_name }}.sha256
+          if-no-files-found: error
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.output_name }}
+          path: metrics/cli/${{ matrix.output_name }}
+          if-no-files-found: error
+
+  cli-publish-actions:
+    name: Metrics CLI Publish Actions
+    needs: [ cli-build-actions ]
+    runs-on: ubuntu-latest
+    env:
+      LINUX_BUILD_NAME: "metrics_cli_linux"
+      MACOS_BUILD_NAME: "metrics_cli_macos"
+      RELEASE_NAME : "metrics-cli-snapshot"
```

```suggestion
      TAG_NAME : "metrics-cli-snapshot"
```
It seems like it is not a release name, but the name of the tag.